cs-verifiable-encryption/README.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: cs-verifiable-encryption
+- Start Date: 2019-07-24
+- RFC PR: 
+- Ursa Issue: 
+- Version: 1
+
+# Summary
+[summary]: #summary
+
+[Camenisch-Shoup verifiable encryption](https://www.shoup.net/papers/verenc.pdf) is a protocol for proving "properties" 
+about encrypted data. The "properties" are discrete log relations and the encryption is public key encryption. The party 
+creating the ciphertext proves in zero knowledge that the ciphertext can be decrypted by the designated party. The RFC 
+talks about 2 things, the encryption scheme itself and a protocol to prove that the encrypted data satisfies specific 
+discrete log relation(s). The CS Verifiable Encryption is the most efficient scheme for verifiable encryption known to the author.
+
+# Motivation
+[motivation]: #motivation
+
+Zero Knowledge Proofs (ZKP) provide anonymity but in some cases the verifier of the ZKP agrees on provisional anonymity meaning 
+that on pre-agreed terms, the anonymity of the prover can be broken. To break this anonymity the verifier takes the proof to a mutually 
+trusted 3rd party who is capable of deanonymizing the prover. Hyperledger Aries will like to use this capability with their anonymous credentials.   
```

Comment:
@dcmiddle Does this statement about "Hyperledger Aries" makes helps in illustrating the importance of the feature?

Reply:
Not really, it does not say what Aries will use it for. The anon creds paper did not mention using this system so it is hard to correlate what is meant by this. In the examples below the first seems like revocation which should be a common feature of all credential systems, i.e. the existing systems should already support revocation without needing this feature.
The second example (linking shipping address) feels artificial. That is subjective and possibly not helpful feedback, but I'm just looking for what is the demand from a Hyperledger project which is not already satisfied by existing Ursa features.

Reply:
> Not really, it does not say what Aries will use it for.

It will be one of the predicates to satisfy while creating proofs from credentials. Like the way we have proof requests that say "reveal attributes a and b, prove c lies in [m, n], prove d is one of these <some set>", there will be an additional "..., prove that you have encrypted attribute e and f for the auditor"

> The anon creds paper did not mention using this system so it is hard to correlate what is meant by this.

I assume you mean for Anoncreds 1.0, that only contains what is implemented (with a small exception). We can add a section for 2.0. But the reference section specifies how it should be used (`encrypt_and_prove_phase_1`, `encrypt_and_prove_phase_2`, `reconstruct_blindings_ciphertext`).

> In the examples below the first seems like revocation

Revocation is not possible until the auditor decrypts the ciphertext. Simply having proof from a credential does not identify the holder, hence the issuer cannot revoke it. If proofs gave clues about the holder (and proof presenter), then the verifier and issuer could collude to deanonymize the holder. 

> The second example (linking shipping address) feels artificial

Can you elaborate on what is artificial? If the shipping company is not controlled by the ecommerce site or for whatever reason does not collude with the ecommerce site, it should not learn about the address. Or are you pointing to the fact that we cannot control whether the buyer is buying for himself or someone else and at that point requiring the address be same from his credential is pointless. Btw, I took that example from the old Idemix site (reference below).

Reply:
Thanks @lovesh the clarifications for Aries usage from you and @dhh1128 are helpful. (The shipping example felt artificial because it did not appear directly related to an Indy/Aries use case).